internal/controller/ai_gateway_route.go

```diff
@@ -427,7 +445,8 @@ func (c *AIGatewayRouteController) newHTTPRoute(ctx context.Context, dst *gwapiv
 			Matches: []gwapiv1.HTTPRouteMatch{
 				{Headers: []gwapiv1.HTTPHeaderMatch{{Name: selectedBackendHeaderKey, Value: key}}},
 			},
-			Filters: rewriteFilters,
+			Filters:  rewriteFilters,
+			Timeouts: dedup[key],
```

Comment:
if the timeout is defined on AIServiceBackend, then we can simply get the timeout value from `b.timeout` in the loop.